core/src/main/java/io/aiven/klaw/service/AclControllerService.java

```diff
@@ -904,4 +907,14 @@ public Set<String> getAivenServiceAccounts(String envId) {
     }
     return serviceAccountsOfTeam;
   }
+
+  public AclRequestsResponseModel getAclRequest(Integer aclRequestId) {
+    String loggedInUser = getCurrentUserName();
+    int tenantId = commonUtilsService.getTenantId(loggedInUser);
+    HandleDbRequests dbHandle = manageDatabase.getHandleDbRequests();
+    AclRequests aclReq = dbHandle.getAcl(aclRequestId, tenantId);
+    aclReq.setEnvironmentName(
+        commonUtilsService.getEnvDetails(aclReq.getEnvironment(), tenantId).getName());
+    return getAclRequestsModels(List.of(aclReq), tenantId, loggedInUser).get(0);
```

Comment:
Is there any chance of an empty list from getAclRequestsModels ?

Reply:
Yes, it is possible I think. Added null check.